test/jdk/java/time/test/java/time/zone/TestZoneRulesProvider.java

```diff
@@ -91,4 +94,48 @@ protected NavigableMap<String, ZoneRules> provideVersions(String zoneId) {
             }
         });
     }
+
+    /**
+     * Tests whether registering a provider twice will still leave it registered.
+     */
+    @Test
+    public void test_registerTwice() {
+        String zone = "MyID";
+        var provider = new ZoneRulesProvider() {
+            @Override
+            protected Set<String> provideZoneIds() {
+                return Set.of(zone);
+            }
+
+            @Override
+            protected ZoneRules provideRules(String zoneId, boolean forCaching) {
+                return null;
+            }
+
+            @Override
+            protected NavigableMap<String, ZoneRules> provideVersions(String zoneId) {
+                return null;
+            }
+        }
+        assertFalse(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected availability for " + zone);
+        ZoneRulesProvider.registerProvider(provider);
+        assertTrue(ZoneId.getAvailableZoneIds().contains(zone), "Unexpected non-availability for " + zone);
+        assertNotNull(ZoneId.of(zone), "ZoneId instance for " + zone + " should be obtainable");
```

Comment:
If the `zone` does not exist, it will not return `null` but throw an exception. Assertion needs to be modified correctly.

Reply:
I just removed the non-`null` assertion as the fact that the call succeeds already is the thing we want to test here.